### DISCUSSION TITLE
Not able to produce dynamic url like "/search/:name"

### DISCUSSION THREAD

**Discussion by hassam-saeed** [vikejs/vike#1388] [Discussion] [answered]
<img width="236" alt="Screenshot 2023-12-27 at 10 48 17 PM" src="https://github.com/vikejs/vike/assets/51261422/21aaa864-0d4b-4bb4-8e7c-c33182e1fa58">
<img width="1050" alt="Screenshot 2023-12-27 at 10 48 35 PM" src="https://github.com/vikejs/vike/assets/51261422/4ddbd41e-4f69-4528-88a3-c97c32a70a6e">

Error:
<img width="1326" alt="Screenshot 2023-12-27 at 10 50 09 PM" src="https://github.com/vikejs/vike/assets/51261422/4b0dece6-8ac7-417c-936a-94f151a6824b">
 
Here I am using the vite-plugin-ssr to convert my react app to SSR react app. 

**Goal**: To make the route dynamic

**Current**: It is giving me the error. Or if error is gone then it is giving me empty object

**Comment by lourot** [comment]
@hassam-saeed we need you to push a minimal reproducer in a GitHub repo. It's very hard to debug anything without having the source-code. Thank you

**Reply by hassam-saeed** [reply]
@AurelienLourot  here is the repo 
https://github.com/hassam-saeed/test-vike-app

I need a route like /search/:name
Here the name is dynamic and I need a name on my Page.JSX component

**Reply by phonzammi** [reply, marked as answer]
Hi @hassam-saeed 
That's is [Route String](https://vike.dev/route-string). The value of name (in your case) is available at pageContext.routeParams.name

To get pageContext in the `+Page` : 

```ts
const pageContext = usePageContext()
```

**Reply by lourot** [reply]
By the way @hassam-saeed would you or your company be up for [sponsoring](https://github.com/sponsors/vikejs/)?
